bluegreen_deploy.py

```diff
@@ -65,228 +67,328 @@ def list_marathon_apps(args):
     return response.json()['apps']
 
 
-def fetch_marathon_app(args, app_ip):
+def fetch_marathon_app(args, app_id):
     response = marathon_get_request(args, "/v2/apps" + app_id)
     return response.json()['app']
 
 
-def get_hostports_from_backends(hmap, backends, haproxy_instance_count):
-    hostports = {}
-    regex = re.compile(r"^(\d+)_(\d+)_(\d+)_(\d+)_(\d+)$", re.IGNORECASE)
-    counts = {}
-    for backend in backends:
-        svname = backend[hmap['svname']]
-        if svname in counts:
-            counts[svname] += 1
-        else:
-            counts[svname] = 1
-        # Are all backends across all instances draining?
-        if counts[svname] == haproxy_instance_count:
-            m = regex.match(svname)
-            host = '.'.join(m.group(1, 2, 3, 4))
-            port = m.group(5)
-            if host in hostports:
-                hostports[host].append(int(port))
-            else:
-                hostports[host] = [int(port)]
-    return hostports
-
-
-def find_tasks_to_kill(tasks, hostports):
-    tasks_to_kill = set()
+def _get_alias_records(hostname):
+    """Return all IPv4 A records for a given hostname
+    """
+    return socket.gethostbyname_ex(hostname)[2]
+
+
+def _unparse_url_alias(url, addr):
+    """Reassemble a url object into a string but with a new address
+    """
+    return urllib.parse.urlunparse((url[0],
+                                    addr + ":" + str(url.port),
+                                    url[2],
+                                    url[3],
+                                    url[4],
+                                    url[5]))
+
+
+def get_marathon_lb_urls(args):
+    """Return a list of urls for all Aliases of the
+       marathon_lb url passed in as an argument
+    """
+    url = urllib.parse.urlparse(args.marathon_lb)
+    addrs = _get_alias_records(url.hostname)
+    return [ _unparse_url_alias(url, addr) for addr in addrs ]
+
+
+def fetch_haproxy_pids(haproxy_url):
+    try:
+        response = requests.get(haproxy_url + "/_haproxy_getpids")
+        response.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        logger.exception("Caught exception when retrieving HAProxy"
+                         " pids from " + haproxy_url)
+        raise e
+
+    return response.text.split()
+
+
+def check_haproxy_reloading(haproxy_url):
+    """Return False if haproxy has only one pid, it is not reloading.
+       Return True if we catch an exception while making a request to
+       haproxy or if more than one pid is returned
+    """
+    try:
+        pids = fetch_haproxy_pids(haproxy_url)
+    except requests.exceptions.RequestException as e:
```

Comment:
This concerns me. It seems like there could be a lot of errors that could come out of this– literally any possible HTTP error– that may cause a "return True" that you didn't intend.